specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2024-06-30/cloudhsm.json

```diff
@@ -0,0 +1,1441 @@
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "The error response describing why the operation failed",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v6/types.json#/definitions/ErrorResponse"
+            },
+            "headers": {
+              "x-ms-request-id": {
+                "description": "A unique ID for the current operation, service generated. All the resource providers must return this value in the response headers to facilitate debugging.",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "CloudHsmCluster_Get_Restore_Status_MaximumSet_Gen": {
+            "$ref": "./examples/CloudHsmCluster_Restore_Pending_MaximumSet_Gen.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CloudHsmProperties": {
+      "type": "object",
+      "description": "The Cloud HSM Properties",
+      "properties": {
+        "fqdn": {
+          "type": "string",
+          "description": "FQDN of the Cloud HSM"
+        },
+        "state": {
+          "type": "string",
+          "description": "The Cloud HSM State"
+        },
+        "stateMessage": {
+          "type": "string",
+          "description": "The Cloud HSM State message"
+        }
+      }
+    },
+    "CloudHsmCluster": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/CloudHsmClusterProperties",
+          "description": "Properties of the Cloud HSM Cluster"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/CloudHsmClusterResource"
+        }
+      ],
+      "description": "Resource information with extended details.",
+      "type": "object"
+    },
+    "CloudHsmClusterResource": {
+      "properties": {
+        "identity": {
+          "$ref": "../../../../../common-types/resource-management/v5/managedidentity.json#/definitions/ManagedServiceIdentity"
+        },
+        "sku": {
+          "$ref": "#/definitions/CloudHsmClusterSku",
+          "description": "SKU details"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "description": "Cloud HSM Cluster Resource",
+      "type": "object",
+      "x-ms-azure-resource": true
+    },
+    "CloudHsmClusterProperties": {
+      "properties": {
+        "activationState": {
+          "type": "string",
+          "description": "State of security domain activation",
+          "readOnly": true
+        },
+        "autoGeneratedDomainNameLabelScope": {
+          "type": "string",
+          "description": "The Cloud HSM Cluster's auto-generated Domain Name Label Scope"
+        },
+        "fipsApprovedMode": {
```

Comment:
>fipsApprovedMode [](http://example.com/codeflow?start=9&length=16)

Recommend enum instead of boolean. This type of property should be modeled with helpful value names, e.g.  ApprovalType: { SingleFactorFIPS, SingleFactor } or something better you come up with. Also enums allow you to provide customer-facing descriptions of each enum value (which is needed to describe FIPS).

Reply:
Based on Customer feedback we made a change and updated this property from a int with Enum values to boolean.